### ISSUE TITLE
Build problem with Python 3

### ISSUE THREAD

**Issue by johannesheinz** [AIDASoft/DD4hep#497]
- **OS:** Ubuntu 18.04
 - **Compiler version:** GCC 8.2.0
 - **DD4hep version:** 01-10
 - **Geant4 version:** 10.05
 - **ROOT version:** git master branch HEAD
 - **Python version:** 3.6.5

I think this issue is related to #252 and #318. At EP-SFT we build the LCG software stack for various operating systems  and Python versions. We discovered a build failure for the configuration Ubuntu 18.04 and Python 3.

Ubuntu + Python2 and CentOS 7/SLC6 + Python3 work fine. I was wondering how the change in #252 would allow Python3 when you specify version 2 explicitly.

Also, it seems, that the `cmake/FindPYTHON.cmake` script is picking up the system Python (2.7.x) instead of our version of Python 3.x that we provide in our build stack and define explicitly as dependency: https://gitlab.cern.ch/sft/lcgcmake/blob/master/projects/CMakeLists.txt#L534

I think there is no configure/build CMake parameter to specify the Python home or Python directory, right?
Python is not in your list of requirements (https://dd4hep.web.cern.ch/dd4hep/usermanuals/DD4hepManual/DD4hepManualch2.html#x3-140002.1.2). Is it necessaty for the build or not?

 - **Input:** https://gitlab.cern.ch/sft/lcgcmake/blob/master/cmake/toolchain/heptools-dev3python3.cmake
 - **Output:** http://cdash.cern.ch/upload/52fab576a537da99fe516f02156bf72e54dbc098/DD4hep-01-10-configure.log
 - **Goal:** I want to build DD4HEP from source on a Ubuntu 18.04 system with Python 3.6.5

I know that Ubuntu is not an officially supported platform, but this behaviour seems a bit strange to me since it is working for other combinations of OS + Python versions. Any input and your experience is apreciated

**Comment by ianna** [comment]
Have you tried to set PYTHON_INCLUDE_DIR to path to where Python.h is found?

**Comment by johannesheinz** [comment]
Thanks for the advice, I think we don't do that automatically. Let me check and I'll report the outcome soon

**Comment by johannesheinz** [comment]
As Dmitri Konstantinov discussed with @MarkusFrankATcernch there's a general build problem with DD4HEP on Python3 systems. We figured out that on our Python3 builds on CentOS 7 and SLC 6, DD4HEP picks up the system Python2 instead of our provided Python3 which leads to a mixed build.

For now, we decided in accordance with the experiments to drop DD4HEP from our Python3 stack until the problem is solved.
